dxr/static/js/code-highlighter.js

```diff
@@ -109,11 +111,23 @@ $(function () {
         }
         if (windowHash) {
             windowHash = windowHash.replace(reCleanup, '');
+            if (permalink.length > 0)
```

Comment:
It looks like it might make sense to factor up replaceState and updatePermalink into a single function. Then we can't forget to update one or the other in the future.